improve shell error handling

fauna-chase opened the pull request:
FE-4795
see above jira for more details

1. shell will not exit on errors
2. shell will correctly reconnect to core in the presence of an intermittent connection error

j

macmv commented on src/lib/fauna-client.ts:
```diff
@@ -57,57 +44,39 @@ export default class FaunaClient {
+    if (res.status === 200 || res.status === 201) {
+      return {
+        status: 200,
+        body: {
+          data: json.data as T,
+        },
```
this should just be
```suggestion
        body: json,
```

so that the `summary` field gets copied over. for example, try this: `fauna eval "log('hi')"`

fauna-chase replied:
o I see, thats why the summary field was shared across error and success types

macmv replied:
yeah. you can also get log messages back in the `summary` for runtime errors too:
```ts
log("hi"); 1/0

// returns

divide_by_zero: Attempted integer division by zero.

error: Attempted integer division by zero.
at *query*:1:14
  |
1 | log('hi'); 1/0
  |              ^
  |

info at *query*:1: hi
```

macmv reviewed (changes requested):
I get some compile errors:
```
src/commands/endpoint/add.ts:123:20 - error TS2339: Property 'close' does not exist on type 'FaunaClient'.

123       await client.close();
                       ~~~~~
```

and my shell crashes after eval:
```
Error: client.close is not a function
    at Object.error (/home/macmv/Desktop/programming/fauna/fauna-shell/node_modules/@oclif/core/lib/errors/index.js:27:15)
    at EvalCommand.error (/home/macmv/Desktop/programming/fauna/fauna-shell/node_modules/@oclif/core/lib/command.js:139:23)
    at EvalCommand.error (/home/macmv/Desktop/programming/fauna/fauna-shell/src/lib/fauna-command.js:49:11)
    at EvalCommand.run (/home/macmv/Desktop/programming/fauna/fauna-shell/src/commands/eval.js:115:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async EvalCommand._run (/home/macmv/Desktop/programming/fauna/fauna-shell/node_modules/@oclif/core/lib/command.js:117:22)
    at async Config.runCommand (/home/macmv/Desktop/programming/fauna/fauna-shell/node_modules/@oclif/core/lib/config/config.js:329:25)
    at async Object.run (/home/macmv/Desktop/programming/fauna/fauna-shell/node_modules/@oclif/core/lib/main.js:89:16)
```

macmv commented on src/lib/fauna-client.ts:
```diff
@@ -57,57 +44,39 @@ export default class FaunaClient {
+    const res = await fetch(url, {
+      method: "POST",
+      headers: {
+        authorization: `Bearer ${secret ?? this.secret}`,
+        "x-fauna-source": "Fauna Shell",
+        ...(typecheck !== undefined && { "x-typecheck": typecheck.toString() }),
+        ...(format !== undefined && { "x-format": format }),
```
the `x-query-timeout-ms` header should be sent here

fauna-chase replied:
good call, looks like we also weren't correctly passing the timeout to the client so resolving that as well.

macmv replied:
oh cool, nice

fauna-chase commented:
yeah my bad, doing my pre commit cleanup I thought I had just removed the old commented out query method but turns out I also removed the non commented out close method.
